tests/recovery/blobmanipulator/blobmanipulator.go

```diff
@@ -0,0 +1,288 @@
+	err := bm.KopiaCommandRunner.ConnectRepo("filesystem", "--path="+bm.DataRepoPath)
+	if err != nil {
+		return "", err
+	}
+
+	if snapID == "" {
+		// list available snaphsots
+		stdout, _, snapshotListErr := bm.KopiaCommandRunner.Run("snapshot", "list", "--json")
+		if snapshotListErr != nil {
+			return stdout, snapshotListErr
+		}
+
+		var s []snapshot.Manifest
+
+		err = json.Unmarshal([]byte(stdout), &s)
+		if err != nil {
+			return "", err
+		}
+
+		// Select a random snapshot to restore
+		snapID = string(s[rand.Intn(len(s))].ID)
+	}
+
+	_, msg, err := bm.KopiaCommandRunner.Run("snapshot", "restore", snapID, restoreDir)
+	if err != nil {
+		// return the error message to parse the object ID to be used in snapshot fix command
+		return msg, err
+	}
+
+	return "", nil
+}
+
+// SetUpSystemUnderTest connects or creates a kopia repo, writes random data in source directory,
+// creates snapshots of the source directory.
+func (bm *BlobManipulator) SetUpSystemUnderTest() error {
+	err := bm.ConnectOrCreateRepo(bm.DataRepoPath)
+	if err != nil {
+		return err
+	}
+
+	// create random data
+	gotFileWriter := bm.getFileWriter()
+	if !gotFileWriter {
+		return errCreatingFileWriter
+	}
+
+	fileSize := 100
+	numFiles := 100
+
+	fileWriteOpts := map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	ctx := context.Background()
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
+
+	// create snapshot of the data
+	snapPath := bm.fileWriter.DataDirectory(ctx)
+	log.Printf("Creating snapshot of directory %s", snapPath)
+
+	_, err = bm.TakeSnapshot(snapPath)
+	if err != nil {
+		return err
+	}
+
+	numFiles = 50
+
+	fileWriteOpts = map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
+
+	fileSize = 40 * 1024 * 1024
+	numFiles = 1
+
+	fileWriteOpts = map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
```

Comment:
Seems these repeated blocks of code could be decomposed into simpler methods to make `SetUpSystemUnderTest` more readable, e.g.
```go
err := bm.writeRandomFiles(fileSize, numFiles)
```